src/mbgl/text/collision_index.cpp

```diff
@@ -415,8 +415,8 @@ std::pair<Point<float>,float> CollisionIndex::projectAndGetPerspectiveRatio(cons
     auto size = transformState.getSize();
     return std::make_pair(
         Point<float>(
-            static_cast<float>(((p[0] / p[3] + 1) / 2) * size.width) + viewportPadding,
-            static_cast<float>(((-p[1] / p[3] + 1) / 2) * size.height) + viewportPadding
+            (((p[0] / p[3] + 1) / 2) * size.width) + viewportPadding,
+            (((-p[1] / p[3] + 1) / 2) * size.height) + viewportPadding
```

Comment:
We have to drop precision at the end of calculation:
```suggestion
            static_cast<float>(((p[0] / p[3] + 1) / 2) * size.width + viewportPadding),
            static_cast<float>(((-p[1] / p[3] + 1) / 2) * size.height + viewportPadding)
```